### ISSUE TITLE
"^" on numbers should be right-associative

### ISSUE THREAD

**Issue by tom111** [Macaulay2/M2#1238]
According [this document](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2-1.15/share/doc/Macaulay2/Macaulay2Doc/html/_precedence_spof_spoperators.html) the operator `^` is left associative. I don't know about the other overloaded types, but for numbers it should be right associative. The following is wrong

    i1 : 3^3^3
    o1 = 19683

The correct answer is

```
i2 : 3^27
o2 = 7625597484987
```

**Comment by DanGrayson** [comment]
I no longer remember why I did it that way, but this looks right:
```

i25 : ZZ^5^6

        30
o25 = ZZ

o25 : ZZ-module, free
```

**Comment by tom111** [comment]
I agree that your example looks right. So this issue cannot be resolved until the underlying issue in Mathematics is solved? Where does Mathematics have its issue tracker?

**Comment by DanGrayson** [comment]
It's probably unusual for someone to every write an expression of the form `A^B^C`, and for readability we might just advise that parentheses always be used.

I'm not sure whether my parser will support it, but we could consider demanding parentheses in such situations, i.e., making the operator non-associative.

**Comment by tom111** [comment]
I really don't know how I would proceed. Maye non-associative "^" is an option, yes.

**Comment by DanGrayson** [comment]
This looks right, too:
```m2
i16 : QQ^6^{3..5}

o16 = | 0 0 0 1 0 0 |
      | 0 0 0 0 1 0 |
      | 0 0 0 0 0 1 |

               3        6
o16 : Matrix QQ  <--- QQ
```

**Comment by DanGrayson** [comment]
This looks right, too:
```m2
i3 : "---" | "a"^1^2 | "---"

        a


o3 = --- ---
```
